amd_openvx_extensions/amd_rpp/source/image/SequenceRearrangeImage.cpp

```diff
@@ -0,0 +1,295 @@
+    STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[6], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+    if (scalar_type != VX_TYPE_UINT32)
+        return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #6 type=%d (must be size)\n", scalar_type);
+    vx_parameter param = vxGetParameterByIndex(node, 1);
+    vx_image image;
+    vx_df_image df_image = VX_DF_IMAGE_VIRT;
+    STATUS_ERROR_CHECK(vxQueryParameter(param, VX_PARAMETER_ATTRIBUTE_REF, &image, sizeof(vx_image)));
+    STATUS_ERROR_CHECK(vxQueryImage(image, VX_IMAGE_ATTRIBUTE_FORMAT, &df_image, sizeof(df_image)));
+    if (df_image != VX_DF_IMAGE_U8 && df_image != VX_DF_IMAGE_RGB)
+        status = VX_ERROR_INVALID_VALUE;
+    STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[1], VX_IMAGE_FORMAT, &df_image, sizeof(df_image)));
+    vx_uint32 height, width;
+    STATUS_ERROR_CHECK(vxQueryImage(image, VX_IMAGE_ATTRIBUTE_HEIGHT, &height, sizeof(height)));
+    STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[1], VX_IMAGE_HEIGHT, &height, sizeof(height)));
+    STATUS_ERROR_CHECK(vxQueryImage(image, VX_IMAGE_ATTRIBUTE_WIDTH, &width, sizeof(width)));
+    STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[1], VX_IMAGE_WIDTH, &width, sizeof(width)));
+    vxReleaseImage(&image);
+    return status;
+}
+
+static vx_status VX_CALLBACK processSequenceRearrangeImage(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+    SequenceRearrangeImageLocalData *data = NULL;
+    vx_status return_status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    vx_df_image df_image = VX_DF_IMAGE_VIRT;
+    STATUS_ERROR_CHECK(vxQueryImage((vx_image)parameters[0], VX_IMAGE_ATTRIBUTE_FORMAT, &df_image, sizeof(df_image)));
+    if (data->device_type == AGO_TARGET_AFFINITY_GPU)
+    {
+        STATUS_ERROR_CHECK(vxQueryImage((vx_image)parameters[0], VX_IMAGE_HEIGHT, &data->dimensions.height, sizeof(data->dimensions.height)));
+        STATUS_ERROR_CHECK(vxQueryImage((vx_image)parameters[0], VX_IMAGE_WIDTH, &data->dimensions.width, sizeof(data->dimensions.width)));
+#if ENABLE_OPENCL
+        cl_command_queue handle = data->handle->cmdq;
+        STATUS_ERROR_CHECK(vxQueryImage((vx_image)parameters[0], VX_IMAGE_ATTRIBUTE_AMD_OPENCL_BUFFER, &data->cl_pSrc, sizeof(data->cl_pSrc)));
+        STATUS_ERROR_CHECK(vxQueryImage((vx_image)parameters[1], VX_IMAGE_ATTRIBUTE_AMD_OPENCL_BUFFER, &data->cl_pDst, sizeof(data->cl_pDst)));
+        unsigned size = data->dimensions.height * data->dimensions.width;
+        if (df_image == VX_DF_IMAGE_U8)
+        {
+            unsigned elem_size = (size / (data->sequence_length * data->sequence_count));
+            for (int sequence_cnt = 0; sequence_cnt < data->sequence_count; sequence_cnt++)
+            {
+                unsigned src_sequence_start_address = sequence_cnt * elem_size * data->sequence_length;
```

Comment:
we generally don't use the address suffix for the variables
please check and remove if possible